Dockerfile

```diff
@@ -146,3 +132,74 @@ RUN if [ -n "$RUN_TESTS" ]; then \
         colcon test-result \
           || ([ -z "$FAIL_ON_TEST_FAILURE" ] || exit 1) \
     fi
+
+# multi-stage for developing
+FROM builder AS dever
+
+# edit apt for caching
+RUN mv /etc/apt/apt.conf.d/docker-clean /etc/apt/
+
+# install developer dependencies
+RUN apt-get update && \
```

Comment:
This doesn't work - you can't move this after the build of the underlay, it could very well easily invalidate that build with dependency library updates.